services/src/main/java/org/apache/unomi/services/impl/rules/RulesServiceImpl.java

```diff
@@ -240,12 +259,23 @@ private void updateRuleStatistics(RuleStatistics ruleStatistics, long ruleCondit
         allRuleStatistics.put(ruleStatistics.getItemId(), ruleStatistics);
     }
 
+    public void refreshRules() {
+        try {
+            allRules = getAllRules();
+        } catch (Throwable t) {
+            logger.error("Error loading rules from persistence back-end", t);
+        }
+    }
+
     private List<Rule> getAllRules() {
         List<Rule> allItems = persistenceService.getAllItems(Rule.class, 0, -1, "priority").getList();
+        Map<String,Set<Rule>> newRulesByEventType = new HashMap<>();
         for (Rule rule : allItems) {
             ParserHelper.resolveConditionType(definitionsService, rule.getCondition(), "rule " + rule.getItemId());
+            updateRulesByEventType(newRulesByEventType, rule);
             ParserHelper.resolveActionTypes(definitionsService, rule);
         }
+        this.rulesByEventType = newRulesByEventType;
```

Comment:
Yes I will remove this, you are right.

Reply:
Still not fan of updating a local internal variable inside this getter.
it would be better to do it inside the refreshRules.

Reply:
I see what you mean but the getAllRules method was private and only used in refreshRules anyway. But in order to prevent any misuse I will refactor this anyway.